mono/metadata/unity-utils.c

```diff
@@ -1976,3 +1976,27 @@ mono_unity_class_has_failure(const MonoClass* klass)
 {
 	return mono_class_has_failure(klass);
 }
+
+#ifdef ANDROID
+static android_network_up_state network_up_state_func = NULL;
+
+MONO_API void
+mono_unity_set_android_network_up_state_func(android_network_up_state func)
+{
+	network_up_state_func = func;
+}
+
+MonoBoolean
+ves_icall_Unity_Android_Network_Interface_Up_State (MonoString *ifName, MonoBoolean* is_up)
+{
+	if (network_up_state_func)
+	{
+		MonoError unused;
+		char* ifNameUtf = mono_string_to_utf8_checked(ifName, &unused);
```

Comment:
I think this string needs freed?

Reply:
Ah yes you're right I missed that when I realized I needed to convert the MonoString to utf8. I'll fix this.